core-rust/mesh-api-server/src/mesh_api/conversions/crypto.rs

```diff
@@ -0,0 +1,57 @@
+use crate::prelude::*;
+use models::SignatureType;
+
+pub(crate) fn extract_public_key(
+    public_key: &crate::mesh_api::generated::models::PublicKey,
+) -> Result<PublicKey, ExtractionError> {
+    match public_key.curve_type {
+        models::CurveType::Secp256k1 => Ok(PublicKey::Secp256k1(
+            hex::decode(&public_key.hex_bytes)
+                .ok()
+                .and_then(|bytes| Secp256k1PublicKey::try_from(bytes.as_slice()).ok())
+                .ok_or(ExtractionError::InvalidSecp256k1PublicKey(
+                    public_key.hex_bytes.clone(),
+                ))?,
+        )),
+        models::CurveType::Edwards25519 => Ok(PublicKey::Ed25519(
+            hex::decode(&public_key.hex_bytes)
+                .ok()
+                .and_then(|bytes| Ed25519PublicKey::try_from(bytes.as_slice()).ok())
+                .ok_or(ExtractionError::InvalidEd25519PublicKey(
+                    public_key.hex_bytes.clone(),
+                ))?,
+        )),
+        t => Err(ExtractionError::InvalidCurveType(t)),
+    }
+}
+
+pub(crate) fn extract_signature(
+    signature: &crate::mesh_api::generated::models::Signature,
+) -> Result<SignatureV1, ExtractionError> {
+    match signature.signature_type {
+        SignatureType::EcdsaRecovery => Ok(SignatureV1::Secp256k1(
+            hex::decode(&signature.hex_bytes)
+                .ok()
+                .and_then(|mut bytes| {
+                    // Mesh uses r + s + v
+                    // Radix uses v + r + s
+                    if let Some(v) = bytes.pop() {
+                        bytes.insert(0, v);
+                    }
+                    Secp256k1Signature::try_from(bytes.as_slice()).ok()
```

Comment:
Has this been tested? I don't think this is quite right. Given their claimed encoding "r (32-bytes) + s (32-bytes) + v (1-byte)", they are probably using default `libsecp256k1` encoding. Which means, compared to Radix, we encode `v + reverse-r + reverse-s` - see our docs here: https://docs.radixdlt.com/docs/concepts-curves-keys-signatures-and-hashing

(Side note - @iamyulong - perhaps we should implement a `from_libsecp256k1_encoding` and `to_libsecp256k1_encoding` in Scrypto?)

Reply:
This is tested by the construction tests.

Reply:
> This is tested by the construction tests.
Ahh right, of course :D

@lrubasze  - maybe it's worth a quick comment explaining that this appears to be yet another different encoding compared to libsecp and radix? :)